### ISSUE TITLE
alias() doesn't support surfacing a *_flag to the command line

### ISSUE THREAD

**Issue by afq984** [bazelbuild/bazel#20582]
### Description of the feature request:

```
load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")

config_setting(
    name = "my_flag_enabled",
    flag_values = {
        ":my_flag": "true",
    },
)

bool_flag(
    name = "my_flag",
    build_setting_default = False,
)

alias(
    name = "my_flag_alias",
    actual = ":my_flag",
)
```

```
bazelisk build --//:my_flag_alias //:my_flag_enabled
INFO: Invocation ID: 4dda7f49-80a0-4ae1-be58-37d1ebd1876d
ERROR: //:my_flag_alias :: Unrecognized option: //:my_flag_alias
```

### Which category does this issue belong to?

Configurability

### What underlying problem are you trying to solve with this feature?

We were trying to expose user-friendly flags aliases to generated repositories, so users don't have to type the full path of packages.

E.g. people can use `--@portage//target/sys-apps/attr:2.5.1_incremental` instead of `--@portage//internal/packages/stage2/target/board/portage-stable/sys-apps/attr:2.5.1_incremental` 

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

development version

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

```
cat .bazelversion 
bbc4b12eb562de5e5daeec9700fa258dfdc5a51b
```

### Any other information, logs, or outputs that you want to share?

For our use case, we also need the `alias`'s `actual` to be in a `select`.

e.g. something like `alias(name = "incremental", actual = select({"//bazel/stage1": "//internal/stage1/pkg/incremental", "//bazel/stage2": "//internal/stage2/pkg/incremental"})`

Not sure if it is something possible to support.

**Comment by afq984** [comment]
I initially confused this with #13463, which was aliasing on the config_setting instead of the bool_flag.

**Comment by gregestren** [comment]
Can https://bazel.build/reference/command-line-reference#flag--flag_alias work for you?

I thought we had some support for what you describe, but I guess not?

**Comment by afq984** [comment]
Our build targets are, generated from a repository rule which inspects local build manifests (which we are slowly  migrating away from).

As the targets are dynamic, we cannot use a static bazelrc with --flag_alias for that.

I think if we write a bazelrc file to be imported by the main file, in the [tools/bazel](https://github.com/bazelbuild/bazelisk?tab=readme-ov-file#toolsbazel) wrapper, then we can get dynamically generated aliases.

It might be hard to pull off though. The build manifest analysis is expensive and lives inside a Bazel invocation already.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
